.ci/integrationTestEC.groovy

```diff
@@ -0,0 +1,198 @@
+    quietPeriod(10)
+  }
+  parameters {
+    string(name: 'BUILD_OPTS', defaultValue: "--no-elasticsearch --no-apm-server --no-kibana --no-apm-server-dashboards --no-apm-server-self-instrument", description: "Addicional build options to passing compose.py")
+  }
+  stages {
+    /**
+     Checkout the code and stash it, to use it on other stages.
+    */
+    stage('Checkout'){
+      steps {
+        deleteDir()
+        gitCheckout(basedir: "${BASE_DIR}")
+        dir("${EC_DIR}"){
+          git(branch: 'master-v2.0',
+            credentialsId: 'f6c7695a-671e-4f4f-a331-acdce44ff9ba',
+            url: 'git@github.com:elastic/observability-test-environments.git'
+          )
+        }
+        stash allowEmpty: true, name: 'source', useDefaultExcludes: false
+      }
+    }
+    stage('Tests On Elastic Cloud'){
+      matrix {
+        agent { label 'linux && immutable' }
+        environment {
+          TMPDIR = "${env.WORKSPACE}"
+          REUSE_CONTAINERS = "true"
+          HOME = "${env.WORKSPACE}"
+          CONFIG_HOME = "${env.WORKSPACE}"
+          EC_WS ="${env.WORKSPACE}/${env.EC_DIR}"
+          VENV = "${env.WORKSPACE}/.venv"
+          PATH = "${env.WORKSPACE}/${env.BASE_DIR}/.ci/scripts:${env.VENV}/bin:${env.EC_WS}/bin:${env.EC_WS}/.ci/scripts:${env.PATH}"
+          CLUSTER_CONFIG_FILE="${env.EC_WS}/tests/environments/elastic_cloud.yml"
+          ENABLE_ES_DUMP = "true"
+        }
+        axes {
+          axis {
+              name 'TEST'
+              values 'all', 'dotnet', 'go', 'java', 'nodejs', 'python', 'ruby', 'rum'
+          }
+          axis {
+              name 'ELASTIC_STACK_VERSION'
+              values '8.0.0-SNAPSHOT', '7.5.1-SNAPSHOT', '7.4.3-SNAPSHOT'
+          }
+        }
+        stages {
+          stage('Prepare Test'){
+            steps {
+              log(level: "INFO", text: "Running tests - ${TEST}")
+              deleteDir()
+              unstash 'source'
+            }
+          }
+          stage('Provision Elastic Cloud environment'){
+            steps {
+              dockerLogin(secret: "${DOCKERELASTIC_SECRET}", registry: "${DOCKER_REGISTRY}")
+              dir("${EC_DIR}/ansible"){
+                withVaultEnv(){
+                  updateEnvConfig()
+                  sh(label: "Deploy Cluster", script: "make deploy-cluster")
+                  archiveArtifacts(allowEmptyArchive: true, artifacts: 'cluster-info/**')
+                }
+              }
+            }
+          }
+          stage("Test") {
+            steps {
+              dir("${BASE_DIR}"){
+                withConfigEnv(){
+                  sh ".ci/scripts/${TEST}.sh"
+                }
+              }
+            }
+          }
+        }
+        post {
+          cleanup {
+            wrappingup("${TEST}")
+            destroyClusters()
+          }
+        }
+      }
+    }
+  }
+  post {
+    cleanup {
+      notifyBuildResult()
+    }
+  }
+}
+
+def withVaultEnv(Closure body){
```

Comment:
This is not a blocker but a todo to be moved to the library

Reply:
Not to forget this, we have this issue: https://github.com/elastic/observability-dev/issues/39